client/cz-cardview.html

```diff
@@ -31,6 +31,118 @@
   </template>
 
   <script>
+    'use strict';
+
+    class CardSpeculation {
+      constructor(columns, height) {
+        this.columns = columns;
+        this.height = height;
+        this.edits = [];
+      }
+
+      insert(column, card, nextCard) {
+        column.height += card._height;
+        var idx = column.cards.indexOf(nextCard);
+        if (idx < 0) // nextCard not found
+          idx = column.cards.length;
+        column.cards.splice(idx, 0, card);
+        this.edits.push({action: 'insert', column: column, card: card, idx: idx});
+      }
+
+      remove(column, card) {
+        column.height -= card._height;
+        var idx = column.cards.indexOf(card);
+        column.cards.splice(idx, 1);
+        this.edits.push({action: 'remove', column: column, card: card});
+      }
+
+      priorities() {
+
+      }
+
+      insertCardBefore(card, nextCard) {
+        var idx = 0;
+        var column = this.columns[idx];
+
+        this.insert(column, card, nextCard);
+        while (idx < this.columns.length) {
+          while (column.height > this.height) {
+            var removedCard = column.cards[column.cards.length - 1];
+            this.remove(column, removedCard);
```

Comment:
This removes cards from the end, then adds to the end of the next column? Might be better to collect all of the overflowing cards in one step, then distribute to columns with remaining space in another to avoid adding/removing the same cards over and over.